components/query_filter/utils.cc

```diff
@@ -117,15 +117,44 @@ static constexpr auto kConditionalQueryStringTrackers =
         {"mkt_tok", "([uU]nsubscribe|emailWebview)"},
     });
 
+// The second parameter is a comma-separated list of domains.
+// The domain comparison will also match on subdomains. So if the
+// parameter is scoped to example.com below, it will be removed from
+// https://example.com/index.php and from http://www.example.com/ for
+// example.
 static constexpr auto kScopedQueryStringTrackers =
     base::MakeFixedFlatMapSorted<std::string_view, std::string_view>({
         // https://github.com/brave/brave-browser/issues/11580
         {"igshid", "instagram.com"},
         // https://github.com/brave/brave-browser/issues/26966
-        {"ref_src", "twitter.com"},
-        {"ref_url", "twitter.com"},
+        {"ref_src", "twitter.com,x.com"},
+        {"ref_url", "twitter.com,x.com"},
+        // https://github.com/brave/brave-browser/issues/34719
+        {"si", "youtube.com,youtu.be"},
     });
 
+bool IsScopedTracker(const std::string_view param_name,
+                     const std::string& spec) {
+  if (kScopedQueryStringTrackers.count(param_name) != 1) {
+    return false;
+  }
+
+  const std::vector<std::string_view> domain_strings =
```

Comment:
frankly i wouldn't premature pessimize - why don't we change `kScopedQueryStringTrackers` to map of string -> std::array or something

Reply:
also given that the complexity slowly grows, it could worth adding a simple unit test

Reply:
> frankly i wouldn't prematurely pessimize - why don't we change `kScopedQueryStringTrackers` to map of string -> std::array or something

Did you mean like this? (note the hard-coded `2` which wastes some memory since it has to be fixed-size)
```
static constexpr auto kScopedQueryStringTrackers =
    base::MakeFixedFlatMapSorted<std::string_view, std::array<std::string_view, 2>>({
        {"igshid", {"instagram.com"}},
        {"ref_src", {"twitter.com", "x.com"}},
    });
```
or converting the fixed-size constexpr map into a dynamic one?

Reply:
@iefremov Is this what you had in mind?
https://github.com/brave/brave-core/pull/21255/commits/5402955023f02e34da937b3cbbbcf76b77fe4842